test/zygote.jl

```diff
@@ -67,6 +68,7 @@ end
             coulomb_const=f32 ? Float32(ustrip(Molly.coulombconst)) : ustrip(Molly.coulombconst),
             force_units=NoUnits,
             energy_units=NoUnits,
+            k = k
```

Comment:
Remove this line, I think this is what's causing the gradient test to fail.

Reply:
I'm not that familar with Zygote but I do not think this error is from my changes:

```
Compiling Tuple{typeof(Molly.convert_k_units), Type{Float64}, Float64, Unitful.FreeUnits{(), NoDims, nothing}}: try/catch is not supported.
  Refer to the Zygote documentation for fixes.
  https://fluxml.ai/Zygote.jl/latest/limitations
```

Reply:
I can look into that after merge.

Reply:
I think it might be the `@assert` macros but commenting them out did not seem to fix the issue.